### PR TITLE
feat: update edit section ui components

### LINKED ISSUE #1099: MuiButtons affected by global styling
The text color of MuiButtons is affected by styles in `global.css` where tailwind properties override settings of Mui:

```css
/* frontend/styles/global.css */
@layer base {
  ...
  a {
    @apply no-underline transition duration-100 cursor-pointer hover:text-accent;
  ...
  }
```

### LINKED ISSUE #1126: Update ui and RSD documentation
We need to update documentation with new mentions approach.
In addition, there are minor ui changes proposed that should be implemented before documentation update.
Include #1099 solution, if not already present.
Include [this remark too](https://github.com/research-software-directory/RSD-as-a-service/commit/2bfb8b8be2da063af0c68ff07bb5ab760d589413#r138657672)

### PR DESCRIPTION
**Pull request by dmijatovic**
# Update ui 

Closes #1099
Closes #1126

Changes proposed in this pull request:
* Use primary button (contained variant) in edit sections
* Use outlined inputs in edit sections
* Remove helper text (not adding a value and make layout more cluttered than needed)
* Changed order in chips input (first selection then selected items)
* Moved ConceptDOI input from left panel to central panel (edit software section). Validate DOI button will automatically update version DOI to concept DOI if possible.
* Update broken tests
* Split edit basic software information into description and links & metadata pages.

How to test:
* `make start` to build app and generate data 
* Add new software and validate working of edit software section
* Check working of validateDOI button
   * use version DOI 10.5281/zenodo.6575742
   * use concept DOI  10.5281/zenodo.6379973
* Add new project and validate working of edit project section
* Login as rsd-admin. Visit organisation and validate working of the edit organisation settings.
* Visit rsd-admin section and validate working of the admin section

## Edit software

- Description
image

- Links & metadata
image

- Contributors
image

- Organisations
image

- Mentions
image

- Mantainers
image

## Edit project
- Details
image

- Mentions
image

## Organisation
- Settings
image

## RSD admin section
image

PR Checklist:

*   [ ] Increase version numbers in `docker-compose.yml`
*   [ ] Link to a GitHub issue
*   [ ] Update documentation
*   [ ] Tests

### CODE REVIEW

**Comment by cmeessen** [comment]
:+1: it's looking good also in Helmholtz style

**Comment by jmaassen** [comment]
Some initial comments (more later): 

Screenshot 2024-02-26 at 16-19-33 Edit software Research Software Directory

Initially, the repository URL box does not contain the "info text". Only after pasting a URL does the text appear: 

Screenshot 2024-02-26 at 16-30-02 Edit software Research Software Directory

Same is true for the "getting started URL". 

Also, the "Platform" info text seems too low? It's not aligned properly with the line?

**Comment by jmaassen** [comment]
Screenshot 2024-02-26 at 16-33-17 Edit software Research Software Directory

The concept DOI section is now really busy, and the button in strangely offset compared to the text box?

**Comment by dmijatovic** [comment]
@jmaassen Thanks for the feedback. I made improvements on software details page.

**Comment by jmaassen** [comment]
I really like the validate button in the text box. Makes it really clear what data it validates.

Some more feedback: 

Screenshot 2024-02-28 at 09-53-39 Edit software Research Software Directory

Could we also move the "keywords" and "licenses" section below the "concept DOI" section? At the moment there is no logic why these are on the side, while the ConceptDOI is in the center. Same is true for log by the way. 

I could also envision that we move these three to a separate tab. The first one could then be called "Description" (now "Software Information", while new one could maybe be called "Metadata" (with DOI, keywords and licenses).

**Comment by jmaassen** [comment]
Regarding the documentation: 

- Overall it looks good. There's some things here and there that can improve, but that's not urgent. 
- Don't we support more package managers than the ones listed in the documentation?

**Comment by dmijatovic** [comment]
@jmaassen, @cmeessen, @ewan-escience After talking to Jason about improvements in the layout and documentation we decided to split ui change and the documentation. This PR handles ui changes. The documentation update will be addressed in the separate PR #1136

**Comment by dmijatovic** [comment]
@cmeessen, @jmaassen I hope this would be the last check to do on this PR :-). 

After some thinking and talking with Jason we decided to simplify software basic info edit page. We divided basic info in two sections.

Can you, please, check again, escpecially edit software section?
Thanks.

File: .env.example [hunk 1]
```diff
@@ -79,12 +79,6 @@ RSD_AUTH_PROVIDERS=SURFCONEXT;ORCID;AZURE;LOCAL
 # Allowed values are: ORCID
 RSD_AUTH_COUPLE_PROVIDERS=ORCID
 
-# Define a semicolon-separated list of user email addresses which are allowed to
-# login to the RSD. If the variable is left empty, or is not defined, all users
-# will be allowed to login.
-# consumed by: authentication
-#RSD_AUTH_USER_MAIL_WHITELIST=user@example.com;test@example.com
-
```
Comment (dmijatovic) [inline .env.example:87]: This env variable is still used. Please set back.

**Comment by jmaassen** [comment]
Screenshot 2024-03-08 at 10-31-18 Edit software Research Software Directory

The subtitle "Where can users find information to start?" now relates to the second URL, not the first. 

Maybe better: 

"Provide links to the source code repository of your software and a webpage describing how to start using it."

**Comment by jmaassen** [comment]
Screenshot 2024-03-08 at 10-31-31 Edit software Research Software Directory

Better subtitle: "Provide the DOI of your software (if available). This DOI will be used to import metadata about the software."

**Comment by jmaassen** [comment]
Screenshot 2024-03-08 at 10-31-45 Edit software Research Software Directory

"Add keywords to your software, or import them using the Software DOI".

**Comment by jmaassen** [comment]
Screenshot 2024-03-08 at 10-31-52 Edit software Research Software Directory

Better text: 

On this page you can provide links and metadata to better describe your software. Some of this metadata can be imported automatically.

__Software URLs__  

Providing the URL of the source code repository of the software allows the RSD show a link to the source code on the software page, and automatically harvest information about the development activity. 

The _getting started URL_ can be used to link to a webpage describing how to install and use the software. This link will be prominently shown on the top of the software page.  

__Software DOI__

The Software DOI is a unique identifier for your software, similar to DOIs for publications or datasets. Using the Software DOI, the RSD can import metadata about the software such as contributors, keywords, and licenses. It also enables the RSD to find new versions of your software, provide citation data to visitors (in various formats such as BibTex, CodeMeta, JATS, RIS and CSL), and track who cites your software. 

You can use [Zenodo](https://help.zenodo.org/faq/#versioning) to create a DOI for your software. Use the __Validate DOI__ button check it the DOI is correct.

__Licences__

Here you can provide the license of your software, by selecting an existing Open Source license, specifying your own, or 
importing the license information from the Software DOI.

__Keywords__

Here you can provide keyword that describe your software, by selecting from existing keyword, adding your own, or importing the keywords from the Software DOI.

**Comment by jmaassen** [comment]
Also: shouldn't we add the contributor info text to that tab instead?

**Review by ewan-escience — APPROVED** [review]

**Pull request merged**